src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -423,6 +507,10 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
 }
 
 enum ReadFuture<S: StateStoreRead> {
+    ReadingInitialPersistedStream {
```

Comment:
May use `StreamExt::Peekable` to merge the two variants on reading persisted stream?

Reply:
Hmm `futures::StreamExt` has some limitations. If we use its peekable implementation, subsequently when we call `peek` we need to `pin_mut`.

I opt for `tokio_stream::StreamExt::peekable` instead. Since it does not have this limitation. Seems like other modules in our codebase (dispatcher) also use this.